pkg/mount/mount_unix_test.go

```diff
@@ -79,6 +81,58 @@ func TestMounted(t *testing.T) {
 	}
 }
 
+func TestMountTmpfsOptions(t *testing.T) {
+	if os.Getuid() != 0 {
+		t.Skip("root required")
+	}
+
+	testCases := []struct {
+		opts       string
+		expected   string
+		unexpected string
+	}{
+		{
+			opts:       "exec",
+			unexpected: "noexec",
+		},
+		{
+			opts:     "noexec",
+			expected: "noexec",
+		},
+	}
+
+	target := path.Join(os.TempDir(), "mount-tmpfs-tests-"+t.Name())
+	if err := os.MkdirAll(target, 0777); err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(target)
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("%s:%q-%q-%q", t.Name(), tc.opts, tc.expected, tc.unexpected), func(t *testing.T) {
+			if err := Mount("tmpfs", target, "tmpfs", tc.opts); err != nil {
+				t.Fatal(err)
+			}
+			defer ensureUnmount(t, target)
+
+			mounts, err := GetMounts(nil)
```

Comment:
Use `GetMounts(SingleEntryFilter(target))` here, it will return a list of 0 or 1 elements so you don't have to loop through those.

Something like
```go
mounts, err := GetMounts(SingleEntryFilter(target))
if err != nil {
        t.Fatal(err)
}
if len(mounts) != 1 {
        t.Fatal("Mount point ", target, " not found")
}
entry := mounts[0]
if !strings.Contains(entry.Opts, tc.expected) {
        t.Fatal("Expected option ", tc.expected, " missing from ", entry.Opts)
}
if tc.unexpected != "" && strings.Contains(entry.Opts, tc.unexpected) {
        t.Fatal("Unexpected mount option ", tc.unexpected, " in ", entry.Opts)
}
```

Reply:
Thank you, fixed.